mono/metadata/domain.c

```diff
@@ -1496,6 +1504,18 @@ mono_context_get (void)
 	return GET_APPCONTEXT ();
 }
 
+gint32
+mono_context_get_id (MonoAppContext *context)
```

Comment:
Lifecycle is better defined in terms of what the user can't expect than on what to expect.

For example, we could say IDs are only unique among live domains/contexts and reuse is unspecified.

Anyway, this can be done after this is merged.